plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/part/AssociationPart.java

```diff
@@ -134,7 +137,8 @@ protected void setConnectionRouting(PolylineConnection conn) {
             if (entityPart == null) {
                 entityPart = getTarget();
             }
-            if (entityPart instanceof GraphicalEditPart) {
+            final DBPPreferenceStore store = ERDUIActivator.getDefault().getPreferences();
+            if (entityPart instanceof GraphicalEditPart && !store.getBoolean(ERDUIConstants.PREF_ROUTING_DIAGRAM_MANHATTAN)) {
```

Comment:
I think we should introduce some sort of decorator interface that will provide all styling info based on the underlying implementation to avoid having all these `if not ... then`